isisd/isis_te.c

```diff
@@ -302,34 +302,83 @@ int isis_mpls_te_update(struct interface *ifp)
 /* Followings are vty command functions */
 #ifndef FABRICD
 
-DEFUN (show_isis_mpls_te_router,
-       show_isis_mpls_te_router_cmd,
-       "show " PROTO_NAME " mpls-te router",
-       SHOW_STR
-       PROTO_HELP
-       MPLS_TE_STR
-       "Router information\n")
-{
+DEFUN(show_isis_mpls_te_router, show_isis_mpls_te_router_cmd,
+      "show " PROTO_NAME " [vrf <NAME|all>] mpls-te router",
+      SHOW_STR PROTO_HELP VRF_CMD_HELP_STR "All VRFs\n"
 
-	struct listnode *anode;
+      MPLS_TE_STR "Router information\n")
+{
+	char *vrf_name = NULL;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+	struct listnode *anode, *nnode, *inode;
 	struct isis_area *area;
+	struct isis *isis = NULL;
 
-	if (!isis) {
+	if (!im) {
 		vty_out(vty, "IS-IS Routing Process not enabled\n");
 		return CMD_SUCCESS;
 	}
-
-	for (ALL_LIST_ELEMENTS_RO(isis->area_list, anode, area)) {
-
-		if (!IS_MPLS_TE(area->mta))
-			continue;
-
-		vty_out(vty, "Area %s:\n", area->area_tag);
-		if (ntohs(area->mta->router_id.s_addr) != 0)
-			vty_out(vty, "  MPLS-TE Router-Address: %s\n",
-				inet_ntoa(area->mta->router_id));
-		else
-			vty_out(vty, "  N/A\n");
+	ISIS_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (vrf_name) {
```

Comment:
Let's find a way to not cut-n-paste code for the case of a single vrf -vs- the default vrf.

Reply:
Sure, will find a way and do it for the case of a single vrf -vs- the default vrf.